cherab/tools/observers/group/plotting.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2016-2021 Euratom
+# Copyright 2016-2021 United Kingdom Atomic Energy Authority
+# Copyright 2016-2021 Centro de Investigaciones Energéticas, Medioambientales y Tecnológicas
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+import matplotlib.pyplot as plt
+from raysect.optical import Spectrum
+from raysect.optical.observer import RadiancePipeline0D, PowerPipeline0D, SpectralRadiancePipeline0D, SpectralPowerPipeline0D
+
+
+def select_pipelines(group, item):
+    """
+    Selects pipelines from the group based on index or name.
+
+    :param Observer0DGroup group: Observer group from which to select pipelines.
+    :param str/int item: The index or name of the pipeline to be selected.
+    :return list pipelines: matching pipelines
+    :return list observers: observers with matching pipelines
+    """
+    pipelines = []
+    observers = []
+    for observer in group._observers:
+        if isinstance(item, int):
+            if item < len(observer.pipelines):
+                pipelines.append(observer.pipelines[item])
+                observers.append(observer)
+        elif isinstance(item, str):
+            matching_pipelines = [pipeline for pipeline in observer.pipelines if pipeline.name == item]
+            if len(matching_pipelines) == 1:
```

Comment:
Ff there is more than 1 matching pipeline, this check will silently fail. Best to invert the logic in the check: if there is not exactly 1 matching pipeline then raise an error.

Reply:
Partially agree, the error should be raised but only if `len(matching_pipelines) > 1`, because in this case we don't know which pipeline to choose. If there is no matching pipeline for this observer, it should be skipped.

Reply:
I have added a warning if no matching pipeline was found. An error is raised only if there are more pipelines.